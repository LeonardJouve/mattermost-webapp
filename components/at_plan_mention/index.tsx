// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import useOpenPricingModal from 'components/common/hooks/useOpenPricingModal';
import useOpenCloudPurchaseModal from 'components/common/hooks/useOpenCloudPurchaseModal';
import useOpenSalesLink from 'components/common/hooks/useOpenSalesLink';

type Props = {
    plan: string;

}

function AtPlanMention(props: Props) {
    const openPricingModal = useOpenPricingModal();
    const openPurchaseModal = useOpenCloudPurchaseModal({});
    const openSalesLink = useOpenSalesLink();

    const handleClick = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
        e.preventDefault();

        if (props.plan === 'Enterprise trial') {
            openPricingModal({trackingLocation: 'notify_admin_message_view'});
        }

        if (props.plan === 'Enterprise plan') {
            openSalesLink();
        }

        if (props.plan === 'Professional plan') {
            openPurchaseModal({trackingLocation: 'notify_admin_message_view'});
        }
    };
    return (
        <a
            onClick={handleClick}
        >
            {props.plan}
        </a>

    );
}

export default AtPlanMention;
