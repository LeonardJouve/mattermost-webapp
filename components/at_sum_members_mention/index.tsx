// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

type Props = {
    text: string;
    userIds: string[];
}

function AtSumOfMembersMention(props: Props) {
    return (
        <a
            onClick={() => {
                console.log('I WAS CLICKED', props.userIds);
            }}
        >
            {props.text}
        </a>

    );
}

export default AtSumOfMembersMention;
