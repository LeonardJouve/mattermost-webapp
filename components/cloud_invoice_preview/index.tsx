// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage} from 'react-intl';

import {useDispatch, useSelector} from 'react-redux';

import {Modal} from 'react-bootstrap';

import PDFPreview from 'components/pdf_preview';

import {closeModal} from 'actions/views/modals';
import {ModalIdentifiers} from 'utils/constants';
import {isModalOpen} from 'selectors/views/modals';
import {GlobalState} from 'types/store';

import './cloud_invoice_preview.scss';

type Props = {
    onHide?: () => void;
    url?: string;
};

function CloudInvoicePreview(props: Props) {
    const dispatch = useDispatch();

    const isPreviewModalOpen = useSelector((state: GlobalState) =>
        isModalOpen(state, ModalIdentifiers.CLOUD_INVOICE_PREVIEW),
    );

    const onHide = () => {
        dispatch(closeModal(ModalIdentifiers.CLOUD_INVOICE_PREVIEW));
        if (typeof props.onHide === 'function') {
            props.onHide();
        }
    };

    return (
        <Modal
            show={isPreviewModalOpen}
            onExited={onHide}
            onHide={onHide}
            id='cloud-invoice-preview'
            className='CloudInvoicePreview'
        >
            <Modal.Header>
                <Modal.Title>{'Invoice'}</Modal.Title>
                <div className={'subtitle'}>
                    {'Download or Print this page for your records'}
                </div>
            </Modal.Header>
            <Modal.Body>
                <div className='cloud_invoice_preview_modal'>
                    <PDFPreview
                        fileInfo={{
                            extension: 'pdf',
                            size: 0,
                            name: 'some-name.pdf',
                        }}
                        fileUrl={
                            'http://localhost:9005/api/v4/files/aphcmr1kjf8qmdqwp6ksqxeier?download=1'
                        }
                        scale={2}
                        handleBgClose={() => {}}
                    />
                </div>
            </Modal.Body>
        </Modal>
    );
}

export default CloudInvoicePreview;
