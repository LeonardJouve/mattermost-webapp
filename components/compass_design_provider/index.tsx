// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React, {FC, memo, ReactNode, useMemo} from 'react';
import {createTheme, ThemeProvider, Theme as MuiTheme} from '@mui/material/styles';

import {Theme} from 'mattermost-redux/selectors/entities/preferences';

interface Props {
    theme?: Theme;
    children?: ReactNode;
}

const CompassDesignProvider: FC<Props> = (props: Props) => {
    const theme = useMemo<MuiTheme>(() => createTheme({
        palette: {
            primary: {
                main: '#166de0',
            },
        },
    }), [props?.theme]);

    return <ThemeProvider theme={theme}>{props.children}</ThemeProvider>;
};

export default memo(CompassDesignProvider);
