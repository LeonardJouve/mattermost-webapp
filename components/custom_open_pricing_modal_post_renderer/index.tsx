// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {useIntl} from 'react-intl';
import {useSelector, useDispatch} from 'react-redux';

import Markdown from 'components/markdown';

import useOpenPricingModal from 'components/common/hooks/useOpenPricingModal';
import {Post} from '@mattermost/types/posts';
import {getUsers} from 'mattermost-redux/selectors/entities/users';
import useOpenCloudPurchaseModal from 'components/common/hooks/useOpenCloudPurchaseModal';
import {openModal} from 'actions/views/modals';
import LearnMoreTrialModal from 'components/learn_more_trial_modal/learn_more_trial_modal';
import {ModalIdentifiers} from 'utils/constants';

const NonAdminPaidFeatures = {
    GUEST_ACCOUNTS: 'Guest Accounts',
    CUSTOM_USER_GROUPS: 'Custom User groups',
    CREATE_MULTIPLE_TEAMS: 'Create Multiple Teams',
    START_CALL: 'Start call',
    PLAYBOOKS_RETRO: 'Playbooks Retrospective',
    UNLIMITED_MESSAGES: 'Unlimited Messages',
    UNLIMITED_FILE_STORAGE: 'Unlimited File Storage',
    ALL_PROFESSIONAL_FEATURES: 'All Professional features',
};

const MinimumPlansForFeature = {
    Professional: 'Professional plan',
    Enterprise: 'Enterprise plan',
};

type FeatureRequest = {
    id: string;
    user_id: string;
    required_feature: string;
    required_plan: string;
    create_at: string;
    trial: string;
}

type RequestedFeature = Record<string, FeatureRequest[]>

type CustomPostProps = {
    requested_features: RequestedFeature;
    trial: boolean;
}

const style = {
    display: 'flex',
    gap: '10px',
    padding: '12px',
    borderRadius: '4px',
    border: '1px solid rgba(var(--center-channel-text-rgb), 0.16)',
    width: 'max-content',
    margin: '10px 0',
};

const btnStyle = {
    background: 'var(--button-bg)',
    color: 'var(--button-color)',
    border: 'none',
    borderRadius: '4px',
    padding: '8px 20px',
    fontWeight: 600,
};

const messageStyle = {
    marginBottom: '16px',
};

export default function OpenPricingModalPost(props: {post: Post}) {
    let allProfessional = true;

    const dispatch = useDispatch();
    const userProfiles = useSelector(getUsers);

    const openPurchaseModal = useOpenCloudPurchaseModal({});
    const {formatMessage} = useIntl();

    const openPricingModal = useOpenPricingModal();

    const postProps = props.post.props as Partial<CustomPostProps>;
    const requestFeatures = postProps?.requested_features;
    const wasTrialRequest = postProps?.trial;

    const customMessageBody = [];
    const getUserIdsForUsersThatRequestedFeature = (requests: FeatureRequest[]): string[] => {
        const userIds: string[] = [];
        requests.forEach((request: FeatureRequest) => {
            userIds.push(request.user_id);
        });

        return userIds;
    };

    const getUserNamesForUsersThatRequestedFeature = (requests: FeatureRequest[]): string[] => {
        const userNames: string[] = [];
        getUserIdsForUsersThatRequestedFeature(requests).forEach((userId: string) => {
            const profile = userProfiles[userId];
            userNames.push('@' + profile?.username);
        });

        return userNames;
    };

    const renderUsersThatRequestedFeature = (requests: FeatureRequest[]) => {
        if (requests.length >= 5) {
            return `${requests.length} members `;
        }

        let renderedUsers;

        const users = getUserNamesForUsersThatRequestedFeature(requests);

        if (users.length === 1) {
            renderedUsers = users[0];
        } else {
            const lastUser = users.splice(-1, 1)[0];
            users.push('and ' + lastUser);
            renderedUsers = users.join(', ').replace(/,([^,]*)$/, '$1');
        }

        return renderedUsers;
    };

    const markDownOptions = {
        atSumOfMembersMentions: true,
        atPlanMentions: true,
        markdown: false,
    };

    const mapFeatureToPlan = (feature: string) => {
        switch (feature) {
        case NonAdminPaidFeatures.GUEST_ACCOUNTS:
        case NonAdminPaidFeatures.CREATE_MULTIPLE_TEAMS:
            return MinimumPlansForFeature.Professional;
        case NonAdminPaidFeatures.CUSTOM_USER_GROUPS:
            allProfessional = false;
            return MinimumPlansForFeature.Enterprise;
        default:
            return MinimumPlansForFeature.Professional;
        }
    };

    if (requestFeatures) {
        for (const featureName of Object.keys(requestFeatures)) {
            const title = (
                <div>
                    <span>
                        <b>
                            {`${featureName}`}
                        </b>
                    </span>
                    <span>
                        <Markdown
                            message={` - available on the ${mapFeatureToPlan(featureName)}`}
                            options={{...markDownOptions, atSumOfMembersMentions: false}}
                        />
                    </span>
                </div>);
            const subTitle = (
                <ul>
                    <li>
                        <Markdown
                            message={`${renderUsersThatRequestedFeature(requestFeatures[featureName])} requested access to this feature`}
                            options={markDownOptions}
                            userIds={getUserIdsForUsersThatRequestedFeature(requestFeatures[featureName])}
                            messageMetadata={{requestedFeature: featureName}}
                        />
                    </li>
                </ul>);

            const featureMessage = (
                <div style={messageStyle}>
                    {title}
                    {subTitle}
                </div>
            );

            customMessageBody.push(featureMessage);
        }
    }

    const openLearnMoreTrialModal = () => {
        dispatch(openModal({
            modalId: ModalIdentifiers.LEARN_MORE_TRIAL_MODAL,
            dialogType: LearnMoreTrialModal,
        }));
    };

    const renderButtons = () => {
        if (wasTrialRequest) {
            return (
                <>
                    <button
                        onClick={openLearnMoreTrialModal}
                        style={btnStyle}
                    >
                        {'Learn more about trial'}
                    </button>
                    <button
                        onClick={openPricingModal}
                        style={{...btnStyle, color: 'var(--button-bg)', background: 'rgba(var(--denim-button-bg-rgb), 0.08)'}}
                    >
                        {formatMessage({id: 'postypes.custom_open_pricing_modal_post_renderer.view_options', defaultMessage: 'View upgrade options'})}
                    </button>
                </>
            );
        }

        if (allProfessional) {
            return (
                <>
                    <button
                        onClick={openPurchaseModal}
                        style={btnStyle}
                    >
                        {'Upgrade to Professional'}
                    </button>
                    <button
                        onClick={openPricingModal}
                        style={{...btnStyle, color: 'var(--button-bg)', background: 'rgba(var(--denim-button-bg-rgb), 0.08)'}}
                    >
                        {formatMessage({id: 'postypes.custom_open_pricing_modal_post_renderer.view_options', defaultMessage: 'View upgrade options'})}
                    </button>
                </>
            );
        }
        return (
            <button
                onClick={openPricingModal}
                style={{...btnStyle, border: '1px solid var(--button-bg)', color: 'var(--button-bg)', background: 'var(--sidebar-text)'}}
            >
                {formatMessage({id: 'postypes.custom_open_pricing_modal_post_renderer.view_options', defaultMessage: 'View upgrade options'})}
            </button>);
    };

    return (
        <div>
            <div style={messageStyle}>
                <Markdown
                    message={props.post.message}
                    options={{...markDownOptions, atSumOfMembersMentions: false}}
                />
            </div>
            {customMessageBody}
            <div style={{display: 'flex'}}>
                <div
                    style={style}
                >
                    {renderButtons()}
                </div>
            </div>
        </div>
    );
}
