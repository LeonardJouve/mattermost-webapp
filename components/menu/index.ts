// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect, ConnectedProps} from 'react-redux';
import {MenuItem, MenuList, Divider} from '@mui/material';

import {getTheme} from 'mattermost-redux/selectors/entities/preferences';

import {GlobalState} from 'types/store';

import {MenuComponent} from './menu';

function mapStateToProps(state: GlobalState) {
    return {
        theme: getTheme(state),
    };
}

const connector = connect(mapStateToProps);

export type PropsFromRedux = ConnectedProps<typeof connector>;

const Menu = connector(MenuComponent);

const MenuDivider = Divider;

export {
    Menu,
    MenuItem,
    MenuList,
    MenuDivider,
};
