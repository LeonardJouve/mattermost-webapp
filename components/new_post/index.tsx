// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';
import {AnyAction, bindActionCreators, Dispatch} from 'redux';

import {showActionsDropdownPulsatingDot} from 'selectors/actions_menu';
import {setActionsMenuInitialisationState} from 'mattermost-redux/actions/preferences';
import {getConfig} from 'mattermost-redux/selectors/entities/general';
import {getPost} from 'mattermost-redux/selectors/entities/posts';

import {
    get,
    isCollapsedThreadsEnabled,
} from 'mattermost-redux/selectors/entities/preferences';
import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
import {getUser} from 'mattermost-redux/selectors/entities/users';

import {Emoji} from '@mattermost/types/emojis';
import {Post} from '@mattermost/types/posts';
import {closeRightHandSide, selectPost} from 'actions/views/rhs';

import {markPostAsUnread, emitShortcutReactToLastPostFrom} from 'actions/post_actions';

import {getShortcutReactToLastPostEmittedFrom, getOneClickReactionEmojis} from 'selectors/emojis';
import {getIsPostBeingEditedInRHS, isEmbedVisible} from 'selectors/posts';
import {getHighlightedPostId} from 'selectors/rhs';
import {getIsMobileView} from 'selectors/views/browser';

import {GlobalState} from 'types/store';

import {isArchivedChannel} from 'utils/channel_utils';
import {areConsecutivePostsBySameUser, shouldShowActionsMenu} from 'utils/post_utils';
import {Preferences} from 'utils/constants';

import PostComponent from './post_component';
import { ExtendedPost, removePost } from 'mattermost-redux/actions/posts';
import { DispatchFunc, GetStateFunc } from 'mattermost-redux/types/actions';
import { isThreadOpen } from 'selectors/views/threads';

interface OwnProps {
    post: Post;
    previousPostId: string;
    teamId: string;
}

function isConsecutivePost(state: GlobalState, ownProps: OwnProps) {
    const post = ownProps.post;
    const previousPost = ownProps.previousPostId && getPost(state, ownProps.previousPostId);

    let consecutivePost = false;

    if (previousPost) {
        consecutivePost = areConsecutivePostsBySameUser(post, previousPost);
    }
    return consecutivePost;
}

function removePostAndCloseRHS(post: ExtendedPost) {
    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
        const state = getState() as GlobalState;
        if (isThreadOpen(state, post.id)) {
            dispatch(closeRightHandSide());
        }
        return dispatch(removePost(post));
    };
}

function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
    const config = getConfig(state);
    const enableEmojiPicker = config.EnableEmojiPicker === 'true';
    const enablePostUsernameOverride = config.EnablePostUsernameOverride === 'true';
    const teamId = ownProps.teamId || getCurrentTeamId(state);
    const channel = state.entities.channels.channels[ownProps.post.channel_id];
    const shortcutReactToLastPostEmittedFrom = getShortcutReactToLastPostEmittedFrom(state);

    const user = getUser(state, ownProps.post.user_id);
    const isBot = Boolean(user && user.is_bot);
    const highlightedPostId = getHighlightedPostId(state);
    const showActionsMenuPulsatingDot = showActionsDropdownPulsatingDot(state);

    let emojis: Emoji[] = [];
    const oneClickReactionsEnabled = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.ONE_CLICK_REACTIONS_ENABLED, Preferences.ONE_CLICK_REACTIONS_ENABLED_DEFAULT) === 'true';
    if (oneClickReactionsEnabled) {
        emojis = getOneClickReactionEmojis(state);
    }

    return {
        enableEmojiPicker,
        enablePostUsernameOverride,
        isEmbedVisible: isEmbedVisible(state, ownProps.post.id),
        isReadOnly: false,
        teamId,
        pluginPostTypes: state.plugins.postTypes,
        channelIsArchived: isArchivedChannel(channel),
        isConsecutivePost: isConsecutivePost(state, ownProps),
        isFlagged: get(state, Preferences.CATEGORY_FLAGGED_POST, ownProps.post.id, null) != null,
        compactDisplay: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.MESSAGE_DISPLAY, Preferences.MESSAGE_DISPLAY_DEFAULT) === Preferences.MESSAGE_DISPLAY_COMPACT,
        colorizeUsernames: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.COLORIZE_USERNAMES, Preferences.COLORIZE_USERNAMES_DEFAULT) === 'true',
        shouldShowActionsMenu: shouldShowActionsMenu(state, ownProps.post),
        showActionsMenuPulsatingDot,
        shortcutReactToLastPostEmittedFrom,
        isBot,
        collapsedThreadsEnabled: isCollapsedThreadsEnabled(state),
        shouldHighlight: highlightedPostId === ownProps.post.id,
        oneClickReactionsEnabled,
        recentEmojis: emojis,
        isExpanded: state.views.rhs.isSidebarExpanded,
        isPostBeingEdited: getIsPostBeingEditedInRHS(state, ownProps.post.id),
        isMobileView: getIsMobileView(state),
    };
}

function mapDispatchToProps(dispatch: Dispatch<AnyAction>) {
    return {
        actions: bindActionCreators({
            markPostAsUnread,
            emitShortcutReactToLastPostFrom,
            setActionsMenuInitialisationState,
            selectPost,
            removePost: removePostAndCloseRHS,
        }, dispatch),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(PostComponent);
