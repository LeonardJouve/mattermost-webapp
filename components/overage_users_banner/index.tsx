// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {FormattedMessage} from 'react-intl';
import {useDispatch, useSelector} from 'react-redux';

import {getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
import {GlobalState} from 'types/store';
import {getLicense} from 'mattermost-redux/selectors/entities/general';
import AlertBanner from 'components/alert_banner';
import {calculateOverageUserActivated} from 'utils/overage_team';
import {isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
import {savePreferences} from 'mattermost-redux/actions/preferences';
import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
import {PreferenceType} from '@mattermost/types/preferences';
import {LicenseLinks, StatTypes, Preferences} from 'utils/constants';

import './overage_users_banner.scss';

type OverageUsersBannerProps = {
    location: 'admin-console' | 'app';
}

type AdminHasWatchItArgs = {
    isAppView: boolean;
    preferenceName: string;
    overagePreferences: PreferenceType[];
    isWarningBanner: boolean;
}

const adminHasWatchIt = ({isAppView, preferenceName, overagePreferences, isWarningBanner}: AdminHasWatchItArgs): boolean => {
    if (isAppView) {
        return overagePreferences.find((value) => value.name === preferenceName) !== undefined;
    }

    if (isWarningBanner) {
        return overagePreferences.find((value) => value.name === preferenceName) !== undefined;
    }

    return false;
};

const OverageUsersBanner = ({location}: OverageUsersBannerProps) => {
    const isAppView = location === 'app';
    const dispatch = useDispatch();
    const stats = useSelector((state: GlobalState) => state.entities.admin.analytics) || {};
    const isAdmin = useSelector(isCurrentUserSystemAdmin);
    const license = useSelector(getLicense);
    const seatsPurchased = parseInt(license.Users, 10);
    const isCloud = useSelector(isCurrentLicenseCloud);
    const getPreferencesCategory = makeGetCategory();
    const currentUser = useSelector((state: GlobalState) => getCurrentUser(state));
    const overagePreferences = useSelector((state: GlobalState) => getPreferencesCategory(state, Preferences.OVERAGE_USERS_BANNER));
    const activeUsers = ((stats || {})[StatTypes.TOTAL_USERS]) as number || 0;
    const {
        isBetween5PercerntAnd10PercentPurchasedSeats,
        isOver10PercerntPurchasedSeats,
    } = calculateOverageUserActivated({
        activeUsers,
        seatsPurchased,
    });
    const prefixPreferences = isOver10PercerntPurchasedSeats ? 'error' : 'warn';
    const preferenceName = `${prefixPreferences}_overage_seats_${seatsPurchased}`;

    const overageByUsers = activeUsers - seatsPurchased;

    const isOverageState = isBetween5PercerntAnd10PercentPurchasedSeats || isOver10PercerntPurchasedSeats;

    if (!isAdmin || !isOverageState || isCloud || adminHasWatchIt({isAppView, isWarningBanner: isBetween5PercerntAnd10PercentPurchasedSeats, overagePreferences, preferenceName})) {
        return null;
    }

    const handleDismiss = () => {
        if (isAppView || isBetween5PercerntAnd10PercentPurchasedSeats) {
            dispatch(savePreferences(currentUser.id, [{
                category: Preferences.OVERAGE_USERS_BANNER,
                name: preferenceName,
                user_id: currentUser.id,
                value: 'Overage users banner watched',
            }]));
        }
    };

    const isDimissable = isAppView || isBetween5PercerntAnd10PercentPurchasedSeats;

    return (
        <AlertBanner
            mode={isOver10PercerntPurchasedSeats ? 'danger' : 'warning'}
            onDismiss={isDimissable ? handleDismiss : undefined}
            title={
                <FormattedMessage
                    id='licensingPage.overageUsersBanner.title'
                    defaultMessage='Your workspace user count has exceeded your paid license seat count by {seats, number} {seats, plural, one {seat} other {seats}}'
                    values={{
                        seats: overageByUsers,
                    }}
                />
            }
            message={
                <FormattedMessage
                    id='licensingPage.overageUsersBanner.description'
                    defaultMessage='Notify your Customer Success Manager on your next true-up check. <a>Contact Sales</a>'
                    values={{
                        a: (chunks: React.ReactNode) => {
                            return (
                                <a
                                    className='overage_users_banner__button'
                                    href={LicenseLinks.CONTACT_SALES}
                                    target='_blank'
                                    rel='noreferrer'
                                >
                                    {chunks}
                                </a>
                            );
                        },
                    }}
                >
                    {(text) => <p className='overage_users_banner__description'>{text}</p>}
                </FormattedMessage>
            }
        />
    );
};

export default OverageUsersBanner;
