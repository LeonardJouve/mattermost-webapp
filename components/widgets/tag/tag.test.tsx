// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';
import {AlertCircleOutlineIcon} from '@mattermost/compass-icons/components';

import Tag from './tag';

const defaultProps = {
    size: 'xs',
    className: 'Tag test',
};

describe('components/widgets/tag/Tag', () => {
    test('should match the snapshot on show', () => {
        const wrapper = shallow(
            <Tag
                className={'test'}
                text={'Test text'}
            />,
        );
        expect(wrapper.props()).toEqual(expect.objectContaining(defaultProps));
        expect(wrapper.text()).toEqual('Test text');
        expect(wrapper).toMatchSnapshot();
    });

    test('should match the snapshot with icon', () => {
        const wrapper = shallow(
            <Tag
                className={'test'}
                text={'Test text'}
                icon={'alert-circle-outline'}
            />,
        );
        expect(wrapper.props()).toEqual(expect.objectContaining(defaultProps));
        expect(wrapper.find(AlertCircleOutlineIcon).exists()).toEqual(true);
        expect(wrapper.text()).toContain('Test text');
        expect(wrapper).toMatchSnapshot();
    });

    test('should match the snapshot with capitalized option', () => {
        const wrapper = shallow(
            <Tag
                className={'test'}
                text={'Test text'}
                capitalize={true}
            />,
        );
        expect(wrapper.props()).toEqual(expect.objectContaining({...defaultProps, capitalize: true}));
        expect(wrapper.text()).toEqual('Test text');
        expect(wrapper).toMatchSnapshot();
    });

    test('should match the snapshot with size "sm"', () => {
        const wrapper = shallow(
            <Tag
                className={'test'}
                text={'Test text'}
                size={'sm'}
            />,
        );
        expect(wrapper.props()).toEqual(expect.objectContaining({...defaultProps, size: 'sm'}));
        expect(wrapper.text()).toEqual('Test text');
        expect(wrapper).toMatchSnapshot();
    });

    test('should match the snapshot with "success" variant', () => {
        const wrapper = shallow(
            <Tag
                className={'test'}
                text={'Test text'}
                variant={'success'}
            />,
        );
        expect(wrapper.props()).toEqual(expect.objectContaining({...defaultProps, className: 'Tag success test'}));
        expect(wrapper.text()).toEqual('Test text');
        expect(wrapper).toMatchSnapshot();
    });

    test('should transform into a button if onClick provided', () => {
        const click = jest.fn();
        const wrapper = shallow(
            <Tag
                className={'test'}
                text={'Test text'}
                onClick={click}
            />,
        );
        expect(wrapper.props()).toEqual(expect.objectContaining({...defaultProps, onClick: click}));
        expect(wrapper.text()).toEqual('Test text');
        wrapper.simulate('click');
        expect(click).toBeCalled();
        expect(wrapper).toMatchSnapshot();
    });

    test('should match the snapshot on hide', () => {
        const wrapper = shallow(
            <Tag
                show={false}
                text={'Test text'}
            />,
        );
        expect(wrapper).toMatchInlineSnapshot('""');
    });
});
