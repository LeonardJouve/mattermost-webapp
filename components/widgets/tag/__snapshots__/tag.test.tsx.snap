// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/tag/Tag should match the snapshot on show 1`] = `
<TagWrapper
  as="div"
  className="Tag test"
  size="xs"
>
  <span
    className="Tag__text"
  >
    Test text
  </span>
</TagWrapper>
`;

exports[`components/widgets/tag/Tag should match the snapshot with "success" variant 1`] = `
<TagWrapper
  as="div"
  className="Tag Tag__success test"
  size="xs"
>
  <span
    className="Tag__text"
  >
    Test text
  </span>
</TagWrapper>
`;

exports[`components/widgets/tag/Tag should match the snapshot with capitalized option 1`] = `
<TagWrapper
  as="div"
  capitalize={true}
  className="Tag test"
  size="xs"
>
  <span
    className="Tag__text"
  >
    Test text
  </span>
</TagWrapper>
`;

exports[`components/widgets/tag/Tag should match the snapshot with icon 1`] = `
<TagWrapper
  as="div"
  className="Tag test"
  size="xs"
>
  <AlertCircleOutlineIcon
    size={10}
  />
  <span
    className="Tag__text"
  >
    Test text
  </span>
</TagWrapper>
`;

exports[`components/widgets/tag/Tag should match the snapshot with size "sm" 1`] = `
<TagWrapper
  as="div"
  className="Tag test"
  size="sm"
>
  <span
    className="Tag__text"
  >
    Test text
  </span>
</TagWrapper>
`;

exports[`components/widgets/tag/Tag should transform into a button if onClick provided 1`] = `
<TagWrapper
  as="button"
  className="Tag test"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  size="xs"
>
  <span
    className="Tag__text"
  >
    Test text
  </span>
</TagWrapper>
`;
